src/app/core/http-services/item-navigation.service.ts

```diff
@@ -88,7 +88,8 @@ export interface NavMenuItem {
   currentScore?: number,
   validated?: boolean,
   canViewContent: boolean,
-  children?: NavMenuItem[] // placeholder for children when fetched (may 'hasChildren' with 'children' not set)
+  children?: NavMenuItem[], // placeholder for children when fetched (may 'hasChildren' with 'children' not set)
+  type?: string,
```

Comment:
Why optional? And then what do you do when you do not have the info?

Reply:
Either the type is given (in which case it has not to be optional), or it is not given and so it means it have to handle it fully handled (have the UI ok).
From what I understand, the type is always given, so you can add it, and there is no reason to make it optional.
(probably better -->) If you annoyed by the `NavMenuItem` type which does not match your needs (it was probably designed for left nav menu only), change the service so that it returns exactly what the service return... and just write a basic wrapper to convert this type to NavMenuItem only for the left menu.

Reply:
Fixed